YubiKit/YubiKit/Helpers/YKFTLVRecord.m

```diff
@@ -0,0 +1,166 @@
+// Copyright 2018-2022 Yubico AB
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <Foundation/Foundation.h>
+#import "YKFTLVRecord.h"
+#import "YKFNSDataAdditions+Private.h"
+
+@interface YKFTLVRecord()
+@property (nonatomic, readwrite) YKFTLVTag tag;
+@property (nonatomic, readwrite) NSData *value;
+@end
+
+@interface NSData (NSData_ByteEncoding)
+    + (NSData *)ykf_dataWithBytesStripLeadingZeros:(char*)bytes length:(int)length;
+@end
+
+@implementation YKFTLVRecord
+
++ (nullable instancetype)recordFromData:(NSData *_Nullable)data checkMatchingLength:(Boolean)checkMatchingLength bytesRead:(int*)bytesRead {
+    *bytesRead = 0;
+    
+    // tag
+    if (data.length == 0) {
+        return nil;
+    }
+    Byte *bytes = (Byte *)data.bytes;
+    int offset = 0;
+    int tag = bytes[offset++] & 0xFF;
+    if ((tag & 0x1F) == 0x1F) {
+        tag = (tag << 8) | (bytes[offset++] & 0xFF);
+        while ((tag & 0x80) == 0x80 && offset < data.length) {
```

Comment:
This will overflow tag silently if it is > than int.max

Reply:
It should be
tag = (tag << 7) | (bytes[offset++] & 0x7F);

Reply:
See:
High-tag-number form. Two or more octets. First octet is as in low-tag-number form, except that bits 5-1 all have value "1." Second and following octets give the tag number, base 128, most significant digit first, with as few digits as possible, and with the bit 8 of each octet except the last set to "1."

Reply:
Both of these two changes result in our implementation producing a different result than Apples implementation in `TKBERTLVRecord`. Not sure what to do.

Reply:
What does the Apple impl seem to do ?